migration/src/jira_util.py

```diff
@@ -289,34 +289,34 @@ def repl_att(m: re.Match):
     return text
 
 
-def embed_gh_issue_link(text: str, issue_id_map: dict[str, str]) -> str:
+def embed_gh_issue_link(text: str, issue_id_map: dict[str, str], gh_number_self: int = None) -> str:
     """Embed GitHub issue number
     """
     def repl_simple(m: re.Match):
         res = m.group(0)
         gh_number = issue_id_map.get(m.group(2))
-        if gh_number:
+        if gh_number and gh_number != gh_number_self:
```

Comment:
Hmm `issue_id_map` is a `str -> str` dict type, but `gh_number` is an `int`?  Aren't we then comparing `str` to `int`?

Reply:
Both `gh_number` and `gh_number_self` is `int`.

Reply:
I fixed type hints in https://github.com/apache/lucene-jira-archive/commit/614a3cd62be8ecf6207e9fb682cd0b376db051fd